.github/workflows/test-fork-prs.yml

```diff
@@ -95,18 +96,104 @@ jobs:
+            const pr_number = fs.readFileSync('./pr_number', 'utf-8');
+            return pr_number
+              .replace(/(\r\n|\n|\r)/gm, '') // remove last new line character
+              .replace(/[^0-9]/g, ''); // remove non-alphanumeric characters
+      - name: Read base sha from artifact
+        uses: actions/github-script@98814c53be79b1d30f795b907e553d8679345975
+        id: read-base-sha
+        with:
+          result-encoding: string
+          script: |
+            const fs = require('fs');
+            const base_sha = fs.readFileSync('./base_sha', 'utf-8');
+            return base_sha
+              .replace(/(\r\n|\n|\r)/gm, '') // remove last new line character
+              .replace(/[^A-Za-z0-9]/g, ''); // remove non-alphanumeric characters
+
       - uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c #v3.3.0 https://github.com/actions/checkout/commit/ac593985615ec2ede58e132d2e21d2b1cbd6127c
       - name: Set status to commit sha
         uses: ./.github/actions/set-status
         with:
-          sha: ${{ steps.read-artifact.outputs.result }}
+          sha: ${{ steps.read-commit-id.outputs.result }}
           state: 'pending'
-          context: 'Run E2E tests'
-          description: 'E2E tests are now running'
+          context: 'Run PR checks'
+          description: 'PR checks are now running'
           # URL below is a link to the current workflow run to allow users to see the status of the workflow.
           target-url: https://github.com/${{ github.event.repository.owner.login }}/${{ github.event.repository.name }}/actions/runs/${{ github.run_id }}
 
+  codeql:
+    needs: setup
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [javascript]
+    steps:
+      - name: Remove run-codeql label, if applicable
+        if: github.event.label.name == 'run-codeql'
+        env:
+          ISSUE_NUMBER: ${{ github.event.pull_request.number }}
+          REPO_OWNER: ${{ github.repository_owner }}
+          REPO_NAME: ${{ github.event.repository.name }}
+          LABEL_NAME: 'run-codeql'
+        uses: actions/github-script@98814c53be79b1d30f795b907e553d8679345975 # v6.4.0 https://github.com/actions/github-script/commit/98814c53be79b1d30f795b907e553d8679345975
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const { ISSUE_NUMBER, REPO_OWNER, REPO_NAME, LABEL_NAME } = process.env
+            github.rest.issues.removeLabel({ owner: REPO_OWNER, repo: REPO_NAME, issue_number: ISSUE_NUMBER, name: LABEL_NAME })
+
+      - name: Checkout
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0 https://github.com/actions/checkout/commit/ac593985615ec2ede58e132d2e21d2b1cbd6127c
+        with:
+          ref: ${{ needs.setup.outputs.commit_id }}
+
+      - name: Initialize CodeQL
```

Comment:
Moving `codeql` to here actually fixes a bug where it didn't have sufficient permission to run on forks.

When workflows runs on forks with `pull_request` trigger, `GITHUB_TOKEN` has [limited scope](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token). We resolved this by moving this to the fork test runner, which has sufficient rights but also runs only when maintainer approves the PR and is not modifiable by contributors.

See https://securitylab.github.com/research/github-actions-preventing-pwn-requests/ for details on this strategy.